hooks/tk-multi-publish2/basic/nuke_update_flame_clip.py

```diff
@@ -126,66 +135,146 @@ def accept(self, settings, item):
+            flame_clip_path = flame_clip_template.apply_fields(flame_clip_fields)
+
+            # Just because we have a template and can build the path where
+            # the clip would exist doesn't mean it's actually there.
+            if os.path.exists(flame_clip_path):
+                item.properties["flame_clip_path"] = flame_clip_path
+        elif item.context.entity:
+            self.logger.debug("Attempting to find OpenClip publish in %s..." % item.context)
+
+            # TODO: We have a problem. The publish2 plugin acceptance routines
+            # are not re-run when the item's link changes. This means if the
+            # context hasn't been set to a shot using the panel app before
+            # publish2 is launched, we end up running this in the project
+            # context, which is never going to work for us here. Because we
+            # then aren't re-run if the user change's the item's link to be
+            # a Shot, we don't look for a clip file when we should have.
+            #
+            # As a fallback to the traditional approach of looking up the clip
+            # path using templates, we now look for a matching PublishedFile
+            # entity. This situation arises when dealing with publishes coming
+            # from Flame 2019+, which makes use of the publish2 app instead of
+            # the old flame export app.
+            #
+            # We're going to do a middle-ground approach when it comes to multiple
+            # clips published to the same shot. If there are multiple clips, we're
+            # going to iterate over each one, newest to oldest, until we find
+            # one that exists locally. Once we find that, we use it.
+            #
+            # In most cases this will not do anything differently than the template
+            # driven solution, because we'll have a single OpenClip published for
+            # the shot, but there might be cases where this heads off a problem.
+            publish_entity_type = get_published_file_entity_type(publisher.sgtk)
+
+            self.logger.debug("Clip publish types to search for: %s" % CLIP_PUBLISH_TYPE)
+
+            clip_publishes = publisher.shotgun.find(
+                publish_entity_type,
+                [
+                    ["entity", "is", item.context.entity],
+                    ["published_file_type.PublishedFileType.code", "is", CLIP_PUBLISH_TYPE],
+                ],
+                fields=(
+                    "path",
+                    "version_number",
+                    "name",
+                    "published_file_type",
+                    "description",
+                ),
+                order=[dict(field_name="id", direction="desc")],
+            )
+
+            if clip_publishes:
+                self.logger.debug("Found published clips: %s" % clip_publishes)
+
+            for clip_publish in clip_publishes:
```

Comment:
This approach is nice because it is trying to gracefully handle the case when the clip file isn't there. But i wonder if we are over thinking it - why not just do a `find_one()`, get the latest publish and if that doesn't point to a valid file, handle that case. btw i am not feeling strongly about this, i just feel that perhaps it's a little bit simpler and therefore better.